staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types_jsonschema.go

```diff
@@ -57,7 +57,7 @@ type JSONSchemaProps struct {
 }
 
 // JSON represents any valid JSON value.
-// These types are supported: bool, int64, float64, string, []interface{}, map[string]interface{} and nil.
+// These types are supported: bool, int64, uint64, float64, string, []interface{}, map[string]interface{} and nil.
```

Comment:
I wouldn't change the external doc here. JSON cannot reliably represent integers higher than 53 bits across clients (many clients treat all JSON numbers as doubles)

Reply:
xref https://github.com/kubernetes/kubernetes/issues/17095

Reply:
But we support that afaik via custom json unmarshaling. @nikhita do you have a link to your first PR?

Reply:
Let me know what I should do please.

Reply:
> But we support that afaik via custom json unmarshaling. @nikhita do you have a link to your first PR?

@sttts https://github.com/kubernetes/kubernetes/pull/44026.

The custom marshalling only preserves/lets use till the highest `int64` number. The custom marshalling was needed because Go's json package converts all numbers to float64 while unmarshalling. This exposed many of the int64 to corruption when it was marshalled back to json.